source/_integrations/tesla_fleet.markdown

```diff
@@ -40,30 +40,140 @@ The Tesla Fleet API {% term integration %} exposes various sensors from Tesla ve
+
+{% warning %}
+The following steps involve sensitive credentials. Never share your `Client Secret` or access token with anyone directly, and ensure you're working in a secure environment.
+{% endwarning %}
+
+1. Get your OAuth details by going to your [Developer dashboard](https://developer.tesla.com/en_US/dashboard), clicking '**View Details**' under the app you set up for Home Assistant integration, then clicking on the '**Credentials & APIs**' tab. Note the `Client ID` and `Client Secret` strings.
+
+2. Run this CURL request, replacing the variable values as specified in the notes below:
+  
+   ```shell
+   CLIENT_ID=REPLACE_THIS_WITH_YOUR_CLIENT_ID
+   CLIENT_SECRET=REPLACE_THIS_WITH_YOUR_CLIENT_SECRET
+   AUDIENCE="https://fleet-api.prd.na.vn.cloud.tesla.com"
+   curl --request POST \
+     --header 'Content-Type: application/x-www-form-urlencoded' \
+     --data-urlencode 'grant_type=client_credentials' \
+     --data-urlencode "client_id=$CLIENT_ID" \
+     --data-urlencode "client_secret=$CLIENT_SECRET" \
+     --data-urlencode 'scope=openid vehicle_device_data vehicle_cmds vehicle_charging_cmds' \
+     --data-urlencode "audience=$AUDIENCE" \
+     'https://fleet-auth.prd.vn.cloud.tesla.com/oauth2/v3/token'
+   ```
+  
+   Notes about the variable values:
+   - For the `CLIENT_SECRET` value, depending on your terminal environment, you may need to escape any `!` and `$` characters in the string, or the curl request will fail.
+   - Replace the `AUDIENCE` value with your region-specific URL. The URL in the example is for users in North America and Asia-Pacific (excluding China). Refer to the [Base URLs documentation](https://developer.tesla.com/docs/fleet-api/getting-started/base-urls) for the URLs for other regions.
+   - For the `scope=...` line, replace the values with a space-delimited list of [the official scope keywords](https://developer.tesla.com/docs/fleet-api/authentication/overview#scopes), as you defined them earlier in your app.
+3. The CURL request should return a response that looks something like:
+  
+   ```json
+   {"access_token":"JSON_WEB_TOKEN","expires_in":28800,"token_type":"Bearer"}
+   ```
+  
+   This is your access token. Copy everything between the double-quotes to be used next.
+4. Run this CURL request, replacing the variable values as specified in the notes below:
+  
+   ```shell
+   curl --location 'https://fleet-api.prd.na.vn.cloud.tesla.com/api/1/partner_accounts' \
+   --header 'Content-Type: application/json' \
+   --header 'Authorization: Bearer JSON_WEB_TOKEN' \
+   --data '{
+       "domain": "my.domain.com"
+   }'
+   ```
+  
+   - If you had to change the `AUDIENCE` URL for your region in step 2, update the main domain of the `--location` arg.
+   - Replace `LONG_GIBBERISH_ACCESS_TOKEN_STRING` with the access token that you copied in the previous step.
```

Comment:
The `LONG_GIBBERISH_ACCESS_TOKEN_STRING` placeholder seems to have been missed during editing.